messaging/src/main/java/org/axonframework/messaging/LambdaBasedMessageTypeResolver.java

```diff
@@ -0,0 +1,168 @@
+    }
+
+    /**
+     * Starts building a new resolver and registers the first payload type with a fixed message type.
+     *
+     * @param payloadType The class to register the fixed message type for
+     * @param messageType The message type to use for the given payload type
+     * @return The type resolver phase for further configuration
+     */
+    public static TypeResolverPhase on(Class<?> payloadType, MessageType messageType) {
+        return on(payloadType, payloadType1 -> messageType);
+    }
+
+    @Override
+    public MessageType resolve(Class<?> payloadType) {
+        var resolver = resolvers.get(payloadType);
+        if (resolver != null) {
+            return resolver.resolve(payloadType);
+        }
+        if (defaultResolver == null) {
+            throw new IllegalArgumentException("No resolver found for payload type [" + payloadType.getName() + "]");
+        }
+        return defaultResolver.resolve(payloadType);
+    }
+
+    /**
+     * Interface representing the phase where type resolvers are registered.
+     */
+    public interface TypeResolverPhase {
+
+        /**
+         * Registers a resolver that returns a {@link MessageType} for the given payload type.
+         *
+         * @param payloadType The class to register the resolver for
+         * @param resolver    The resolver that produces a {@link MessageType} for the given payload type
+         * @return The current phase for further configuration
+         * @throws IllegalArgumentException if a resolver is already registered for the given payload type
+         */
+        TypeResolverPhase on(Class<?> payloadType, MessageTypeResolver resolver);
+
+        /**
+         * Registers a fixed {@link MessageType} for the given payload type.
+         *
+         * @param payloadType The class to register the fixed message type for
+         * @param messageType The message type to use for the given payload type
+         * @return The current phase for further configuration
+         * @throws IllegalArgumentException if a resolver is already registered for the given payload type
+         */
+        default TypeResolverPhase on(Class<?> payloadType, MessageType messageType) {
+            return on(payloadType, __ -> messageType);
+        }
+
+        /**
+         * Configures the resolver to throw an exception when no specific resolver is found for a payload type. This is
+         * the default behavior.
+         *
+         * @return The completed {@link LambdaBasedMessageTypeResolver}
+         */
+        LambdaBasedMessageTypeResolver onUnknownFail();
+
+        /**
+         * Configures the resolver to use the specified default resolver when no specific resolver is found for a
+         * payload type.
+         *
+         * @param resolver The default resolver to use
+         * @return The completed {@link LambdaBasedMessageTypeResolver}
+         */
+        LambdaBasedMessageTypeResolver onUnknownUse(MessageTypeResolver resolver);
```

Comment:
I think I would rename this method to `fallback`. Since it is the fallback solution, right?